<?xml version="1.0"?>
<launch>

<!--参数-->

    <!-- run pointcloud_to_laserscan node -->
    <node pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node" name="pointcloud_to_laserscan">
 
        <remap from="cloud_in" to="/scan2"/>
        
        <rosparam>
            # target_frame: rslidar # Leave disabled to output scan in pointcloud frame
            transform_tolerance: 0.01
            min_height: 0.1
            max_height: 1.0
 
            angle_min: -3.1415926 # -M_PI
            angle_max: 3.1415926 # M_PI
            angle_increment: 0.01 # 0.57degree
            scan_time: 0.1
            range_min: 0.2
            range_max: 100
            use_inf: true
            inf_epsilon: 1.0
 
            # Concurrency level, affects number of pointclouds queued for processing and number of threads used
            # 0 : Detect number of cores
            # 1 : Single threaded
            # 2->inf : Parallelism level
            concurrency_level: 1
        </rosparam>
 
    </node>

<!--地图文件yaml-->
<arg name="map_file" default="$(find sentry_laser_sim)/map/rm_map1.yaml"/>



<!--运行map_server节点,加载地图-->
<node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)">
    <param name="frame_id" value="map"/>
</node>

<!--运行amcl节点-->
    <include file="$(find amcl)/examples/amcl_omni.launch">
    </include>  

    <!-- 对于虚拟定位，需要设置一个/odom与/map之间的静态坐标变换 -->
    <!--node pkg="tf" type="static_transform_publisher" name="map_odom_broadcaster" args="0 0 0 0 0 0 /map /odom 100" /-->

    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
	<param name="controller_frequency" value="10.0"/> 
    	<param name="controller_patiente" value="15.0"/>
    	<rosparam file="$(find sentry_laser_sim)/config/navigation/costmap_comman_params_amcl.yaml" command="load" ns="global_costmap"/>
    	<rosparam file="$(find sentry_laser_sim)/config/navigation/costmap_comman_params_amcl.yaml" command="load" ns="local_costmap"/>

    	<rosparam file="$(find sentry_laser_sim)/config/navigation/local_costmap_params.yaml" command="load"/>
    	<rosparam file="$(find sentry_laser_sim)/config/navigation/global_costmap_params.yaml" command="load"/>

    	<rosparam file="$(find sentry_laser_sim)/config/navigation/base_local_planner_params.yaml" command="load"/>
    </node>
<param name="/amcl/initial_pose_x"            value="0.0"/>
<param name="/amcl/initial_pose_y"            value="0.0"/>
<param name="/amcl/initial_pose_a"            value="0.0"/>
<node name="rviz" pkg="rviz" type="rviz" args="-d $(find sentry_laser_sim)/rviz/nav_amcl.rviz" required="true" /> 


</launch>
